src/backend/utils/storage.js

```diff
@@ -26,31 +26,34 @@ module.exports = {
    * Feeds
    */
   addFeed: async (feed) => {
-    const key = createFeedKey(feed.id);
-    await redis
-      .multi()
-      // Using hmset() until hset() fully supports multiple fields:
-      // https://github.com/stipsan/ioredis-mock/issues/345
-      // https://github.com/luin/ioredis/issues/551
-      .hmset(
-        key,
-        'id',
-        feed.id,
-        'author',
-        feed.author,
-        'url',
-        feed.url,
-        'user',
-        feed.user,
-        'link',
-        feed.link,
-        'etag',
-        feed.etag,
-        'lastModified',
-        feed.lastModified
-      )
-      .sadd(feedsKey, feed.id)
-      .exec();
+    // Check if feed being added already exists in flagged feeds set
+    if (!(await redis.sismember(flaggedFeedsKey, feed.id))) {
```

Comment:
This is probably right, but one thing it will do is show that a feed was added successfully, but in reality we'll skip adding it.  Should we error?  I'm not sure we should.  But this will be a kind of "silent" no-op/failure.

Reply:
what if we return the redis operation values?

For example when you try to add a Feed with `addFeed(feed)` and it is successful
![image](https://user-images.githubusercontent.com/18711727/80399268-ee1b8e80-8886-11ea-932f-cceaa21d4380.png)

vs adding a Feed that exists in `t:feeds:flagged` set
![image](https://user-images.githubusercontent.com/18711727/80399279-f378d900-8886-11ea-8e94-149c65511dd7.png)